corehq/apps/toggle_ui/static/toggle_ui/js/flags.js

```diff
@@ -18,6 +18,9 @@ hqDefine('toggle_ui/js/flags', [
                 return true;
             }
             var tag = aData[0].replace(/\n/g," ").replace(/<.*?>/g, "");
+            if (viewModel.tagFilter() === "Solutions" && tag.includes("Solutions")) {
```

Comment:
Is this just looking for the string "Solutions" within strings like "Solutions - Open Use"? If so maybe it would be better to have something like `tag_group="solutions"` on each of these new solutions tags, and then check that field here. I think that'd make the logic here more explicit and robust, instead of relying on the human readable display name for the tag. This is minor, but I thought I'd make the suggestion

Reply:
I definitely agree that there should be a more robust way to check this. Other than adding a global variable instead of checking `viewModel.tagFilter() === "Solutions"`, how could I do this? Maybe I'm misunderstanding how toggles work on this page, but `viewModel.tagFilter()` and `tag` are both just strings (the name of the tag).

Reply:
Yeah I see what you mean. I dug a bit and can see that the toggles are used in the django template to create an html table, and the toggle JSON isn't passed to JS, which limits what kind of logic you can do cleanly here. So maybe I'd just keep this in mind and if the logic for this page starts getting more complicated and messy, it might be worth revisiting.

For the time being, a quick thing that could prevent bugs arising from someone innocently renaming the solutions sub-tags would be to add a test for `toggles.py` that just asserts that the sub tags start with the literal string `"Solutions -"`, and add comments to the test pointing to the logic that depends on that. That way if someone changes the string it'll break the build.

Reply:
Great point. Thank you!